File: sdk/javascript/src/resources/analytics.ts
import type { RecoEngineClient } from "../client";
import type { FeedbackSummary, OverviewStats, UsageStats } from "../types";

export class Analytics {
  constructor(private readonly client: RecoEngineClient) {}

  /** Item count, query volume today and this month, average latency, top items. */
  async overview(): Promise<OverviewStats> {
    return (await this.client.request<OverviewStats>({ method: "GET", url: "/analytics/overview" })).data;
  }

  /** Feedback counts by type for the last `days` days (default 30). */
  async feedbackSummary(options: { days?: number } = {}): Promise<FeedbackSummary> {
    const response = await this.client.request<FeedbackSummary>({
      method: "GET",
      url: "/analytics/feedback-summary",
      params: { days: options.days },
    });
    return response.data;
  }

  /** Daily volume, query types and cache hit rate for the last `days` days (default 30). */
  async usage(options: { days?: number } = {}): Promise<UsageStats> {
    const response = await this.client.request<UsageStats>({
      method: "GET",
      url: "/analytics/usage",
      params: { days: options.days },
    });
    return response.data;
  }
}

File: sdk/python/recoengine/resources/analytics.py
"""Usage analytics."""

from __future__ import annotations

from typing import TYPE_CHECKING

from recoengine.models import FeedbackSummary, OverviewStats, UsageStats

if TYPE_CHECKING:
    from recoengine.client import AsyncRecoEngineClient, RecoEngineClient


class Analytics:
    def __init__(self, client: RecoEngineClient) -> None:
        self._client = client

    def overview(self) -> OverviewStats:
        """Item count, query volume today and this month, average latency, top items."""
        return OverviewStats.model_validate(
            self._client._request("GET", "/analytics/overview").json()
        )

    def feedback_summary(self, *, days: int = 30) -> FeedbackSummary:
        response = self._client._request(
            "GET", "/analytics/feedback-summary", params={"days": days}
        )
        return FeedbackSummary.model_validate(response.json())

    def usage(self, *, days: int = 30) -> UsageStats:
        response = self._client._request("GET", "/analytics/usage", params={"days": days})
        return UsageStats.model_validate(response.json())


class AsyncAnalytics:
    def __init__(self, client: AsyncRecoEngineClient) -> None:
        self._client = client

    async def overview(self) -> OverviewStats:
        response = await self._client._request("GET", "/analytics/overview")
        return OverviewStats.model_validate(response.json())

    async def feedback_summary(self, *, days: int = 30) -> FeedbackSummary:
        response = await self._client._request(
            "GET", "/analytics/feedback-summary", params={"days": days}
        )
        return FeedbackSummary.model_validate(response.json())

    async def usage(self, *, days: int = 30) -> UsageStats:
        response = await self._client._request("GET", "/analytics/usage", params={"days": days})
        return UsageStats.model_validate(response.json())
